src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -633,6 +636,32 @@ class TaskBuilderTest extends MarathonSpec {
     assert("1001" == env("PORT_8081"))
   }
 
+  test("PortsEnvWithCustomPrefix") {
+    val command =
+      TaskBuilder.commandInfo(
+        AppDefinition(
+          ports = Seq(8080, 8081)
+        ),
+        Some(TaskID("task-123")),
+        Some("host.mega.corp"),
+        Seq(1000, 1001),
+        Some("CUSTOM_PREFIX_")
+      )
+    val env: Map[String, String] =
+      command.getEnvironment.getVariablesList.asScala.toList.map(v => v.getName -> v.getValue).toMap
+
+    assert(!env.contains("PORT"))
+    assert("1000" == env("CUSTOM_PREFIX_PORT"))
+
+    assert(!env.contains("PORT0"))
+    assert("1000" == env("CUSTOM_PREFIX_PORT0"))
+    assert("1000" == env("CUSTOM_PREFIX_PORT_8080"))
+
+    assert(!env.contains("PORT1"))
+    assert("1001" == env("CUSTOM_PREFIX_PORT1"))
+    assert("1001" == env("CUSTOM_PREFIX_PORT_8081"))
+  }
```

Comment:
Explicitly check for all Variables not having the prefix and compare it against a list of exceptions.

That's more future proof in my opinion.